activation/activation.go

```diff
@@ -619,27 +633,52 @@ func (b *Builder) PublishActivationTx(ctx context.Context) error {
 func (b *Builder) createAtx(ctx context.Context) (*types.ActivationTx, error) {
 	b.log.With().Info("challenge ready")
 
-	pubEpoch := b.challenge.PublishEpoch()
-
 	hash, err := b.challenge.Hash()
 	if err != nil {
 		return nil, fmt.Errorf("getting challenge hash failed: %w", err)
 	}
 
-	// the following method waits for a PoET proof, which should take ~1 epoch
-	expireLayer := (pubEpoch + 2).FirstLayer()
-	atxExpired := b.layerClock.AwaitLayer(expireLayer) // this fires when the target epoch is over
+	pubEpoch := b.challenge.PublishEpoch()
+	// Calculate deadline for waiting for poet proofs.
+	// Deadline must fir between current poet round end and
+	// start of the next. It must also accommodate for PoST duration.
+	//                               PoST
```

Comment:
the block for PoST should be after the deadline? (i.e. you need the proof as input to do PoST)